### PR TITLE
Add workspace support when no packages are specified

### PR DESCRIPTION
**Pull request by JLugagne**
In the current version of go (v1.20.2), there is no way to easily test all the modules of a workspace.
To handle this issue, I have added a small piece of code in the resolvePackages function to look at the current folder to know which type of project it is: if a go.work file exists but no go.mod then the function will parse the file and retrieve all the known module.

This PR handle projects with the following structure:
```myworkspace/
  go.work
  mod1/
    go.mod
    file.go
  mod2/
    go.mod
    file.go
```

Please, feel free to ask for refactoring if needed.

### CODE REVIEW

File: gocov/test.go [hunk 1]
```diff
@@ -32,26 +33,53 @@ import (
...
+		if err := goWork.Close(); err != nil {
+			panic(err)
+		}
+		workFile, err := modfile.ParseWork("go.work", content, nil)
+		if err != nil {
+			return resolvedPkgs, err
+		}
+		for _, use := range workFile.Use {
+			resolvedPkgs = append(resolvedPkgs, use.Path)
+		}
```
Comment (axw) [inline gocov/test.go:66]: In this case, the `pkgs` parameter is completely ignored. That seems like surprising behaviour to me.

Also, the values you're appending to `resolvedPkgs` are module paths, not package paths. A module may have multiple packages within it, and the top level module path does not necessarily have any Go source, and will thus not be a valid package path.

**Review by axw — COMMENTED** [review]
Thanks for opening the PR. There's a couple of issues with the logic, which I've commented on inline.

I think this will do what you want?

```bash
PACKAGES=$(go list -f '{{if and .Module .Module.Main}}{{.ImportPath}}{{end}}' all)
gocov test $PACKAGES
```

**Comment by JLugagne** [comment]
Thank you for your review. It seems to solve what I'm trying to solve. Do you think it could make sense to have an argument to do that in gocov? Or to run this when test on "./..." is asked?

**Comment by axw** [comment]
@JLugagne I don't think this is something that should be done in gocov. If there's a common need for listing all packages in all modules of a workspace, then that seems like something that should be added to `go list`.

**Pull request closed without merging**